clamor/gateway/emitter.py

```diff
@@ -1,28 +1,199 @@
 # -*- coding: utf-8 -*-
```

Comment:
Can we move this whole file to a `utils` package, instead of `gateway`? As it doesn't contain any specific code pertaining to gateway, so it would be good to generally make it reusable, in the library as well as for end-users who want similar functionality for features in their bot.

Reply:
That makes sense, I'll make the change